src/reader/buffered_reader.rs

```diff
@@ -117,7 +117,7 @@ macro_rules! impl_buffered_source {
                     // somewhere sane rather than at the EOF
                     Ok(n) if n.is_empty() => return Err(bang_type.to_err()),
                     Ok(available) => {
-                        if let Some((consumed, used)) = bang_type.parse(buf, available) {
+                        if let Some((consumed, used)) = bang_type.parse(&buf[start..], available) {
```

Comment:
I think, I understood. We should parse only newly read data from the `buf`. This is correct, because we in the function that parses the whole comment, so any data that exists in the buffer before is not relevant here and should be skipped.

Reply:
Yes, exactly. I have added the comment. Not doing so was making some checks like [the one on this line](https://github.com/tafia/quick-xml/blob/60249aeea24bfe00848a70950ae7cd321cacfb3c/src/reader/mod.rs#L828) to be incorrect.